File: reverseInteger.py
import sys

def reverse(a):
    if a == -sys.maxint-1:
        raise Exception("cannot reverse "+str(a))

    if a < 0:
        return - _reverse(-a)
    else:
        return _reverse(a)

def _reverse(a):
    r = 0
    while a > 0:
        digit = a % 10
        a = a / 10

        if sys.maxint / 10 >= r and sys.maxint - digit >= r*10:
            r = r * 10 + digit
        else:
            raise Exception('overflow!')

    return r


if __name__ == '__main__:
    print reverse(123)

File: ReverseInteger.java
public class ReverseInteger{
  public static int reverse(int x){
    //get the last digit and append it to the end of the new value
    if(x == Integer.MIN_VALUE){
      //overflow
      return 0;
    }

    if( x < 0){
      return - reverse(-x);
    }

    int result = 0;

    while(x > 0){
      int lastDigit = x % 10;

      if(result <= Integer.MAX_VALUE / 10){
        result = result * 10 + lastDigit;
      }else{
        return 0;
      }


      x = x / 10;
    }
    return result;
  }

  public static void main(String[] args){
    System.out.println(ReverseInteger.reverse(100));
  }
}
